From f60e328afeaa0ea668f3bf70d06585e58851fe80 Mon Sep 17 00:00:00 2001
From: Xiaoyao Li <xiaoyao.li@intel.com>
Date: Thu, 17 Feb 2022 15:35:04 +0800
Subject: [PATCH 1116/1418] x86/cpu/tdx: Validate p_seamldr_ready flag

to see if a valid P-SEAMLDR loaded.

Signed-off-by: Xiaoyao Li <xiaoyao.li@intel.com>
---
 arch/x86/kernel/cpu/tdx/p-seamldr.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/x86/kernel/cpu/tdx/p-seamldr.c b/arch/x86/kernel/cpu/tdx/p-seamldr.c
index fef647957a45..4c44c6a51320 100644
--- a/arch/x86/kernel/cpu/tdx/p-seamldr.c
+++ b/arch/x86/kernel/cpu/tdx/p-seamldr.c
@@ -432,6 +432,12 @@ static int __init p_seamldr_get_info(void)
 	if (err)
 		goto out;
 
+	if (!p_seamldr_info->p_seamldr_ready) {
+		pr_err("p_seamldr_ready is not set, it seems buggy P-SEAMLDR\n");
+		err = -EINVAL;
+		goto out;
+	}
+
 	pr_info("TDX P-SEAMLDR: version 0x%0x attributes 0x%0x vendor_id 0x%x "
 		"build_date %d build_num 0x%x minor 0x%x major 0x%x.\n",
 		p_seamldr_info->version, p_seamldr_info->attributes,
-- 
2.31.1

